src/main/java/gov/usgs/wma/waterdata/collections/CollectionParams.java

```diff
@@ -36,4 +42,51 @@
 		return paramMap;
 	}
 
+	public String getServerUrl() {
+		return serverUrl;
+	}
+
+	public int getMaxLimit() {
+		return maxLimit;
+	}
+
+	public void setLimit(Map<String, Object> params, int limit) {
+		params.put("limit",limit);
+	}
+
+	public void setStartIndex(Map<String, Object> params, int startIndex) {
+		params.put("startIndex",startIndex);
+	}
+
+	/*
+	 * a bounding box with four numbers:
+     * Lower left corner,  longitude
+     * Lower left corner,  latitude
+     * Upper right corner, longitude
+     * Upper right corner, latitude
+	 */
+	public void setBbox(Map<String, Object> params, List<Double> bbox) {
+		if (bbox != null && bbox.size() == 4) {
+			params.put("pointLowLeft", String.format("Point(%f %f)", bbox.get(0), bbox.get(1)));
+			params.put("pointUpRight", String.format("Point(%f %f)", bbox.get(2), bbox.get(3)));
+		}
+	}
+
+	public void setPagingParams(Map<String, Object> params, int count) {
+		int startIndex = params.get("startIndex") == null ? START_INDEX_DEFAULT : (Integer) params.get("startIndex");
+		int limit = params.get("limit") == null ? maxLimit : (Integer) params.get("limit");
+
+		if (startIndex > 0) {
+			int prevStartIndex = startIndex - limit;
+			if (prevStartIndex < 0) {
+				prevStartIndex = 0;
+			}
+			params.put("prevStartIndex", String.format("&startIndex=%d&limit=%d", prevStartIndex, limit));
+		}
+
+		int nextStartIndex = startIndex + limit;
+		if (nextStartIndex <= count - 1) {
+			params.put("nextStartIndex", String.format("&startIndex=%d&limit=%d", nextStartIndex, limit));
+		}
```

Comment:
Ed does not see my comment. Testing if this reply makes it available to him.

Reply:
I think maybe it should be:
int nextStartIndex = Math.min(count, startIndex + limit);
we would still need the if check on line 88
they seem about equivalent.

Reply:
you would if you do not subtract one from count in the min. if you subtract one then it would be  complete. I would test it of course and make truth tables to make sure they are equal.

Reply:
true, at this point I would prefer to leave as is for now.